src/test/java/org/opensearch/knn/integ/BinaryIndexIT.java

```diff
@@ -0,0 +1,242 @@
+import org.junit.BeforeClass;
+import org.opensearch.client.Request;
+import org.opensearch.client.Response;
+import org.opensearch.client.ResponseException;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.knn.KNNMappingsBuilder;
+import org.opensearch.knn.KNNResult;
+import org.opensearch.knn.TestUtils;
+import org.opensearch.knn.index.VectorDataType;
+import org.opensearch.knn.index.query.KNNQueryBuilder;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.knn.common.Constants.FIELD_FILTER;
+import static org.opensearch.knn.common.Constants.FIELD_TERM;
+import static org.opensearch.knn.common.KNNConstants.ENCODER_SQ;
+import static org.opensearch.knn.common.KNNConstants.K;
+import static org.opensearch.knn.common.KNNConstants.KNN;
+import static org.opensearch.knn.common.KNNConstants.MAX_DISTANCE;
+import static org.opensearch.knn.common.KNNConstants.METHOD_HNSW;
+import static org.opensearch.knn.common.KNNConstants.MIN_SCORE;
+import static org.opensearch.knn.common.KNNConstants.PATH;
+import static org.opensearch.knn.common.KNNConstants.QUERY;
+import static org.opensearch.knn.common.KNNConstants.TYPE_NESTED;
+import static org.opensearch.knn.common.KNNConstants.VECTOR;
+
+/**
+ * This class contains integration tests for binary index with HNSW in Faiss
+ */
+@Log4j2
+public class BinaryIndexIT extends AutoIndexCleanUpTestCase {
+    private static TestUtils.TestData testData;
+
+    @BeforeClass
+    public static void setUpClass() throws IOException {
+        if (BinaryIndexIT.class.getClassLoader() == null) {
+            throw new IllegalStateException("ClassLoader of BinaryIndexIT Class is null");
+        }
+        URL testIndexVectors = BinaryIndexIT.class.getClassLoader().getResource("data/test_vectors_binary_1000x128.json");
+        URL testQueries = BinaryIndexIT.class.getClassLoader().getResource("data/test_queries_binary_100x128.csv");
+        URL groundTruthValues = BinaryIndexIT.class.getClassLoader().getResource("data/test_ground_truth_binary_100.csv");
+        assert testIndexVectors != null;
+        assert testQueries != null;
+        assert groundTruthValues != null;
+        testData = new TestUtils.TestData(testIndexVectors.getPath(), testQueries.getPath(), groundTruthValues.getPath());
+    }
+
+    @SneakyThrows
+    public void testFaissHnswBinary_whenKnnIndexIsTrue_thenCreateIngestQueryWorks() {
+        // Create Index
+        String indexName = getCurrentMethodName().toLowerCase(Locale.ROOT);
+        String fieldName = "my-vector";
+        createKnnHnswBinaryIndex(true, KNNEngine.FAISS, indexName, fieldName, 16, null);
+
+        // Ingest
+        Byte[] vector1 = { 0b00000001, 0b00000001 };
```

Comment:
wondering why we did not use the resources that are in setup in setupClass() for this one

Reply:
The resource in setup is large(1000) compared to this one.
In this test, I wanted to validate the query result with specific order of doc ids which is not easy with 1000 data.